cmd/hamctl/command/actions/artifact_id.go

```diff
@@ -0,0 +1,58 @@
+package actions
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	httpinternal "github.com/lunarway/release-manager/internal/http"
+)
+
+func ArtifactIDFromEnvironment(client *httpinternal.Client, service, namespace, environment string) (string, error) {
+	var statusResp httpinternal.StatusResponse
+	params := url.Values{}
+	params.Add("service", service)
+	if namespace != "" {
+		params.Add("namespace", namespace)
+	}
+	path, err := client.URLWithQuery("status", params)
+	if err != nil {
+		return "", err
+	}
+	err = client.Do(http.MethodGet, path, nil, &statusResp)
+	if err != nil {
+		return "", err
+	}
+
+	switch environment {
+	case "dev":
+		return statusResp.Dev.Tag, nil
+	case "staging":
+		return statusResp.Staging.Tag, nil
+	case "prod":
+		return statusResp.Prod.Tag, nil
+	}
+
+	return "", fmt.Errorf("unknown environment %s", environment)
+}
+
+func ArtifactIDFromBranch(client *httpinternal.Client, service string, branch string) (string, error) {
+	var describeResp httpinternal.DescribeArtifactResponse
+	params := url.Values{}
+	params.Add("branch", branch)
+	path, err := client.URLWithQuery(fmt.Sprintf("describe/latest-artifact/%s/%s", service, branch), params)
+	if err != nil {
+		return "", err
+	}
+	err = client.Do(http.MethodGet, path, nil, &describeResp)
+	if err != nil {
+		return "", err
+	}
+
+	if len(describeResp.Artifacts) == 0 {
+		// TODO: write better message:
+		return "", fmt.Errorf("no artifacts found on master branch")
```

Comment:
How about something like this? I think that will suffice for now?

```suggestion
		return "", fmt.Errorf("no artifact found from branch '%s'", branch)
```

This will give this experience:

```
$ hamctl promote --env dev
Error: no artifact found from branch 'master'
```